### ISSUE TITLE
pre signed url is not accessible after 15 mins, throws clock skew error

### ISSUE THREAD

**Issue by anandhu-karattu** [noobaa/noobaa-core#8490]
### Environment info
- NooBaa Version: **noobaa-core-5.17.0-20241023.el9.x86_64**
- Platform: **standalone noobaa**

### Actual behavior
pre signed url creation is success and accessible via browser as well as curl command.
However after 15 mins (approximately) , the url becomes inaccessible and gives clockskew error.

### Expected behavior
If I set the expiry time of the pre signed url, the url must be accessible till that expiry time.

### Steps to reproduce
1. Upload a text file with some readable data in to bucket . 
[root@redshirt1 ~]# cat test123.txt 
abcd
[root@redshirt1 ~]# aws-test s3 cp test123.txt s3://bucket1001

2. Generate a pre signed url with expiration time of 7 days
[root@redshirt1 ~]# aws-test s3 presign s3://bucket1001/test123.txt --expires-in 604800

3. Now try to read the url via curl command or browser , Getting read properly.
Noobaa.log shows the following entry

```
Oct 30 02:45:36 enfield-x-master [1997121]: [nsfs/1997121]   [L0] core.sdk.bucketspace_fs:: BucketSpaceFS.read_bucket_sdk_info: bucket name bucket1001
Oct 30 02:45:36 enfield-x-master [1997121]: [nsfs/1997121]   [L0] core.sdk.bucketspace_fs:: BucketSpaceFS.read_bucket_sdk_info: bucket name bucket1001
Oct 30 02:45:36 enfield-x-master [1997121]: [nsfs/1997121]   [L0] core.sdk.namespace_fs:: NamespaceFS: read_object_stream { file_path: '/ibm/fs1/account1001/bucket1001/test123.txt', start: 0, end: Infinity, size: 5 }
Oct 30 02:45:36 enfield-x-master [1997121]: [nsfs/1997121]   [L0] core.sdk.namespace_fs:: NamespaceFS: read_object_stream completed file /ibm/fs1/account1001/bucket1001/test123.txt { num_bytes: 5, num_buffers: 1, avg_buffer: 5, log2_size_histogram: { '3': 1 } }

```

4. After roughly 15 mins do the same operations in step3 again.

`Oct 30 03:08:47 enfield-x-master [1997120]: [nsfs/1997120] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>RequestTimeTooSkewed</Code><Message>The difference between the request time and the server&apos;s time is too large.</Message><Resource>/bucket1001/test123.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&amp;X-Amz-Credential=CoQUAQeCnOBNmROQxmuf%2F20241030%2Fus-east-1%2Fs3%2Faws4_request&amp;X-Amz-Date=20241030T094528Z&amp;X-Amz-Expires=604800&amp;X-Amz-SignedHeaders=host&amp;X-Amz-Signature=e09fada28fccfbb058cf235d45758bb119a4d5ffaa317cc39f5ae137d85951a4</Resource><RequestId>m2vps331-2q0kh1-ndv</RequestId></Error> GET /bucket1001/test123.txt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=CoQUAQeCnOBNmROQxmuf%2F20241030%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241030T094528Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=e09fada28fccfbb058cf235d45758bb119a4d5ffaa317cc39f5ae137d85951a4 {"host":"9.30.217.60:6443","connection":"keep-alive","cache-control":"max-age=0","sec-ch-ua":"\"Chromium\";v=\"130\", \"Google Chrome\";v=\"130\", \"Not?A_Brand\";v=\"99\"","sec-ch-ua-mobile":"?0","sec-ch-ua-platform":"\"macOS\"","upgrade-insecure-requests":"1","user-agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7","sec-fetch-site":"none","sec-fetch-mode":"navigate","sec-fetch-user":"?1","sec-fetch-dest":"document","accept-encoding":"gzip, deflate, br, zstd","accept-language":"en-US,en;q=0.9,ms;q=0.8,ml;q=0.7"} Error: The difference between the request time and the server's time is too large.    at Object.check_headers (/usr/local/noobaa-core/src/util/http_utils.js:602:15)    at handle_request (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:106:16)    at Object.s3_rest [as handler] (/usr/local/noobaa-core/src/endpoint/s3/s3_rest.js:66:15)    at Server.endpoint_request_handler (/usr/local/noobaa-core/src/endpoint/endpoint.js:286:28)    at Server.emit (node:events:518:28)    at Server.emit (node:domain:488:12)    at parserOnIncoming (node:_http_server:1143:12)    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)`

**Comment by anandhu-karattu** [comment]
@romayalon @guymguym Can you please look in to this?

**Comment by guymguym** [comment]
Our check in the code is indeed not adapted to the presigned url case correctly - it checks the request date is up to fixed 15 minutes (config.AMZ_DATE_MAX_TIME_SKEW_MILLIS) -

https://github.com/noobaa/noobaa-core/blob/0b9e0d1cb89f048a24f1043a5c41876c6ee2cf55/src/util/http_utils.js#L590-L603

AWS docs limit the [presigned URL's to a maximum of 7 days](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#PresignedUrl-Expiration):

>A presigned URL remains valid for the period of time specified when the URL is generated. If you create a presigned URL with the Amazon S3 console, the expiration time can be set between 1 minute and 12 hours. If you use the AWS CLI or AWS SDKs, the expiration time can be set as high as 7 days.

I think we should fix to have a separate config for the limit of presigned requests, so for regular signed requests (aka headers_v4) we indeed check the 15 minutes (config.AMZ_DATE_MAX_TIME_SKEW_MILLIS) to avoid time skew of the client and server, but for presigned requests (aka query_v4) we will check for 7 days (e.g config.AMZ_DATE_MAX_PRESIGNED_MILLIS).

Note that we already [check the expiration date itself during the signature verification](https://github.com/noobaa/noobaa-core/blob/0b9e0d1cb89f048a24f1043a5c41876c6ee2cf55/src/util/signature_utils.js#L127-L133), so no need to change that.

**Comment by guymguym** [comment]
cc @nimrod-becker @romayalon

**Comment by romayalon** [comment]
I merged a fix to master #8493 , closing this issue
